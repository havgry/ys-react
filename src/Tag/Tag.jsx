import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Tag = ({
  label,
  className,
  ...props
}) => {
  const wrapperProps = {
    className: classNames(
      'ys-tag',
      className,
    ),
    ...props,
  }

  return (
    <span
      {...wrapperProps}
    >
      {label}
    </span>
  )
}

Tag.propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
}

Tag.defaultProps = {
  className: '',
}

export default Tag
