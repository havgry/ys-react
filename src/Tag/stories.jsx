import React from 'react'
import { storiesOf } from '@storybook/react'

import { text } from '@storybook/addon-knobs'

import Tag from './Tag'

storiesOf('Tag', module)
  .addParameters({ jest: ['Tag'] })
  .add('Default', () => (
    <Tag
      label={text('label', 'Nyhed', 'Content')}
      className={text('className', '', 'Misc')}
    />
  ))
