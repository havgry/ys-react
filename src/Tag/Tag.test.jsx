import React from 'react'
import { shallow, mount as mountComponent } from 'enzyme'

import Tag from './Tag'

const defaultLabel = 'Nyhed'
const customClassName = 'panda'
const getTag = props => (
  <Tag
    label={defaultLabel}
    {...props}
  />
)

const shallowRender = props => shallow(getTag(props))
const mount = props => mountComponent(getTag(props))

describe('<Tag>', () => {
  describe('Props', () => {
    it('Throws propType warning if no `label` prop is passed', () => {
      const component = () => shallowRender({
        label: null,
      })
      expect(component).toThrow()
    })

    it('Has correct defaultProps', () => {
      const component = mount()

      const {
        className,
      } = component.props()

      expect(className).toBe('')
    })
  })

  describe('Rendering', () => {
    it('Outputs elements with the correct class names', () => {
      const component = shallowRender()
      expect(component.find('span.ys-tag').exists()).toBe(true)
    })

    it('Custom `className` is being applied to the container', () => {

      const component = shallowRender({
        className: customClassName,
      })

      expect(component.find(`.ys-tag.${customClassName}`).exists()).toBe(true)
    })
  })

  describe('Snapshots', () => {
    it('With custom `className`', () => {
      const component = shallowRender({
        className: customClassName,
      })

      expect(component).toMatchSnapshot()
    })

    it('Without custom `className`', () => {
      const component = shallowRender()

      expect(component).toMatchSnapshot()
    })
  })
})
