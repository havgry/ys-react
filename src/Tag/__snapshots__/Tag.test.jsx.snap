// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag> Snapshots With custom \`className\` 1`] = `
<span
  className="ys-tag panda"
>
  Nyhed
</span>
`;

exports[`<Tag> Snapshots Without custom \`className\` 1`] = `
<span
  className="ys-tag"
>
  Nyhed
</span>
`;
